test/test_torch.py

```diff
@@ -2620,6 +2620,26 @@ def test_index_add(self):
             dest2[idx[i]] = dest2[idx[i]] + src[i]
         self.assertEqual(dest, dest2)
 
+    # add coverage for issue with atomic add that appeared only for 
+    # specific dtypes on cuda:
+    # https://github.com/pytorch/pytorch/issues/29153
+    def test_dtypes_for_index_add(self):
```

Comment:
I wouldn't call this "test_dtypes_for_index_add". The other test doesn't test dtypes, but it really should (on the other hand, `test_index_add` shouldn't test CUDA because then it would run for too long).

What you're doing here is checking that an index_add between zeros and a tensor gives back the original tensor. How about naming this something like `test_index_add_trivial` or something

Reply:
I don't like "trivial" because it doesn't capture what's actually tested. The main purpose of this test is to capture if index_add doesn't work for some dtypes due to special dtype handling code. I could do another PR at some point to merge the two though.

Reply:
Again, I feel really weird having both tests. In terms of design, it makes sense for all tests in test_torch, unless otherwise specified, to test all dtypes; something named "test_index_add" should really test everything unless there are multiple tests that have a prefix named "test_index_add" that have differentiating suffixes. I don't want to block this fix on that but it would be great if you could send a followup PR to clean the tests up.

Reply:
I'm fine to eventually merge the two. I agree all of these tests should eventually be running for all dtypes.